### DISCUSSION TITLE
Conveyor can produce invalid MSIX

### DISCUSSION THREAD

**Discussion by mwhapples** [hydraulic-software/conveyor#104] [Q&A] [answered]
I am not sure whether this is a conveyor bug or a limitation of MSIX. It appears that when I include a file with an uppercase file extension (eg. myimage.PNG) the resulting MSIX installer is invalid and when I attempt to run it I get an error (message shown in the installer window is "Error in parsing the app pakage."). I am not sure whether this is a limitation of MSIX or a bug in how conveyor builds the MSIX package, however I would at least expect conveyor to issue a warning or error for such a problem file rather than just silently making the invalid MSIX. When I remap the input to give it a lowercase extension (IE. myimage.png) the error goes away and the MSIX seems to be valid. I should be able to work around this limitation, but getting notified should such files get in would be good to have.

**Comment by mikehearn** [comment]
This is a bug in Conveyor and we will get it fixed. Thanks for the report!

**Comment by mwhapples** [comment]
Thanks for this. Could you give me an indication of what the fix will be? I mean, is it worth me spending time to fix the file names to only have lowercase extensions, will uppercase extensions be permitted after the fix or will conveyor just notify users of the issue?

**Reply by mikehearn** [reply]
It should be a few line fix. Uppercase extensions will be allowed after. How much work is it for you to work around the issue for now? We just did a release, but can probably get a test build with a fix to you in a few days or by next week.

**Comment by mwhapples** [comment]
We are switching from Java's jpackage to conveyor, at the moment we have not yet made a release with conveyor and do not have a release pending. In short we can wait a little time for the fix.

**Reply by mikehearn** [reply]
We have a fix for this and it'll go into the next release.

**Reply by mwhapples** [reply]
Thanks for this fix, looking forward to the next release. Also more widely thanks for conveyor, it looks like it should solve so many of our app distribution problems we were having with JPackage.

**Reply by mikehearn** [reply, marked as answer]
We believe this has been fixed in Conveyor 14.1

**Comment by mwhapples** [comment]
Thanks, just tried it and it all seems to work as I would hope/expect with the app I am bundling.
